### PR TITLE
Improve Typescript in `BasePlatform`

### PR DESCRIPTION
**Pull request by SimonBrandner**
Towards https://github.com/vector-im/element-web/issues/22235

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/BasePlatform.ts [hunk 1]
```diff
@@ -107,8 +107,7 @@ export default abstract class BasePlatform {
      * Update the currently running app to the latest available version
      * and replace this instance of the app with the new version.
      */
-    installUpdate() {
-    }
+    public abstract installUpdate(): void;
```
Comment (weeman1337) [inline src/BasePlatform.ts:110]: This method was not `abstract` before.

File: src/BasePlatform.ts [hunk 2]
```diff
@@ -365,10 +368,6 @@ export default abstract class BasePlatform {
         window.location.href = mxClient.getSsoLoginUrl(callbackUrl.toString(), loginType, idpId); // redirect to SSO
     }
 
-    onKeyDown(ev: KeyboardEvent): boolean {
```
Comment (weeman1337) [inline src/BasePlatform.ts:368]: Did you delete this method by intent?

**Review by weeman1337 — COMMENTED** [review]
Hey :wave: Looks good. Can you double-check my two comments if these changes are okay? They are more than defining visibilities like mentioned in the issue description.

File: src/BasePlatform.ts (hunk 2 above)
Comment (SimonBrandner) [reply src/BasePlatform.ts:368]: It doesn't seem to be used anywhere - yes, it was intentional

File: src/BasePlatform.ts (hunk 1 above)
Comment (SimonBrandner) [reply src/BasePlatform.ts]: Oops - sorry

**Review by weeman1337 — APPROVED** [review]

**Pull request merged**